logstash-core/lib/logstash/runner.rb

```diff
@@ -184,6 +198,21 @@ def configure_plugin_paths(paths)
     end
   end
 
+  def agent_class
+    return @agent_class if @agent_class # We don't want to load these things twice
```

Comment:
I do agree with eliminating the method and I would like to make the exception part of the `create_agent` method.